### PR TITLE
Fix: Employee Mapping Error

### PR DESCRIPTION
**Pull request by ruuushhh**
## Summary by CodeRabbit

- **Bug Fixes**
	- Improved handling of vendor and employee IDs in journal entry creation to ensure accurate mapping based on brand ID.
	- Enhanced validation logic for employee mapping to better manage error handling and assignment when entities are not found.
	- Added checks to prevent potential errors when accessing properties of the entity variable.

- **New Features**
	- Introduced a new function for verifying employee mapping, centralizing logic and improving error handling for employee mapping validations.
	- Added a new function to retrieve employee or vendor mappings, enhancing clarity and reducing redundancy.
	- Added new tests to validate employee mapping functionality and ensure proper exception handling.

### CODE REVIEW

File: apps/sage_intacct/tasks.py [hunk 1]
```diff
@@ -359,31 +359,31 @@ def __validate_employee_mapping(expense_group: ExpenseGroup, configuration: Conf
...
-                    if configuration.employee_field_mapping == 'EMPLOYEE':
-                        entity = entity.destination_employee
-                    else:
-                        entity = entity.destination_vendor
-
-                    if not entity:
-                        raise EmployeeMapping.DoesNotExist
-
-            elif configuration.corporate_credit_card_expenses_object == 'EXPENSE_REPORT':
+        elif configuration.corporate_credit_card_expenses_object == 'JOURNAL_ENTRY':
```
Comment (ashwin1111) [inline apps/sage_intacct/tasks.py:362]: why are we checking only for ccc. Can we do config.reimburse == je or config.ccc  == je

Also, we have this block in models,
```
if lineitem.fund_source == 'CCC' and configuration.use_merchant_in_journal_line and lineitem.vendor:
    vendor = DestinationAttribute.objects.filter(attribute_type='VENDOR', value__iexact=lineitem.vendor, workspace_id=expense_group.workspace_id).order_by('-updated_at').first()
    if vendor:
        vendor_id = vendor.destination_id
```
we should escape validation for this case

File: apps/sage_intacct/tasks.py [hunk 2]
```diff
@@ -359,31 +359,31 @@ def __validate_employee_mapping(expense_group: ExpenseGroup, configuration: Conf
...
+                if configuration.employee_field_mapping == 'EMPLOYEE':
+                    entity = entity.destination_employee
+                else:
+                    entity = entity.destination_vendor
 
-                if not entity.destination_employee:
+                if not entity:
                     raise EmployeeMapping.DoesNotExist
 
+        elif configuration.corporate_credit_card_expenses_object == 'EXPENSE_REPORT':
```
Comment (ashwin1111) [inline apps/sage_intacct/tasks.py:377]: 347 update condition to `if expense_group.fund_source == 'PERSONAL' or (expense_group.fund_source == 'CCC' 
 and config.ccc == 'ER')

Comment (ashwin1111) [reply apps/sage_intacct/tasks.py:377]: if that is done, this complete block is not required

File: apps/sage_intacct/tasks.py [hunk 3]
```diff
@@ -342,47 +356,22 @@ def __validate_employee_mapping(expense_group: ExpenseGroup, configuration: Conf
...
-                        workspace_id=expense_group.workspace_id
-                    )
-                    if configuration.employee_field_mapping == 'EMPLOYEE':
-                        entity = entity.destination_employee
-                    else:
-                        entity = entity.destination_vendor
-
-                    if not entity:
-                        raise EmployeeMapping.DoesNotExist
+        if expense_group.fund_source == 'CCC' and configuration.use_merchant_in_journal_line:
```
Comment (ashwin1111) [inline apps/sage_intacct/tasks.py:362]: and ccc = JE

Comment (ashwin1111) [reply apps/sage_intacct/tasks.py:362]: and brand id = Fyle

File: apps/sage_intacct/tasks.py [hunk 4]
```diff
@@ -342,47 +356,22 @@ def __validate_employee_mapping(expense_group: ExpenseGroup, configuration: Conf
...
-                entity = EmployeeMapping.objects.get(
-                    source_employee__value=expense_group.description.get('employee_email'),
-                    workspace_id=expense_group.workspace_id
-                )
+        if expense_group.fund_source == 'PERSONAL' or (expense_group.fund_source == 'CCC' and configuration.corporate_credit_card_expenses_object == 'EXPENSE_REPORT'):
+            verify_employee_mapping(expense_group, configuration)
 
-                if not entity.destination_employee:
-                    raise EmployeeMapping.DoesNotExist
+        elif configuration.corporate_credit_card_expenses_object == 'JOURNAL_ENTRY' or configuration.reimbursable_expenses_object == 'JOURNAL_ENTRY':
```
Comment (ashwin1111) [inline apps/sage_intacct/tasks.py:372]: we already have 1 round of JE validation at top, club required conditions along with that

**Review by ashwin1111 — APPROVED** [review]

**Review by Hrishabh17 — APPROVED** [review]

**Pull request merged**